sdk/ai/ai-inference-rest/README.md

```diff
@@ -0,0 +1,313 @@
+# Azure Inference REST client library for JavaScript
+
+Inference API for Azure-supported AI models
+
+**Please rely heavily on our [REST client docs](https://github.com/Azure/azure-sdk-for-js/blob/main/documentation/rest-clients.md) to use this library**
+
+Key links:
+
+- [Source code](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/openai/openai-rest)
+- [Package (NPM)](https://aka.ms/npm-azure-rest-ai-inference)
+- [API reference documentation](https://aka.ms/AAp1kxa)
+- [Samples](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/openai/openai/samples)
+
+## Getting started
+
+```javascript
+import ModelClient from "@azure-rest/ai-inference";
+import { AzureKeyCredential } from "@azure/core-auth";
+const client = new ModelClient(
+  "https://<Azure Model endpoint>",
+  new AzureKeyCredential("<Azure API key>")
+);
+
+const response = await client.path("/chat/completions").post({
+  body: {
+    messages: [
+      {role: "user", content: "How many feet are in a mile?"},
+    ],
+  }
+});
+
+if(response.status !== "200") {
```

Comment:
could we utilize the helper method `isUnexpected()`? https://github.com/Azure/azure-sdk-for-js/pull/29863/files#diff-c0a1f87972cf51790d6074c463516039f4809c9fdbb87e9abf652d881a61dca5R311